test/unit/bibDetailsUtils.test.js

```diff
@@ -251,4 +255,82 @@ describe('bibDetailsUtils', () => {
       expect(constructSubjectHeadingsArray(url)).to.deep.equal(expectedResult);
     });
   });
+
+  describe('isRtl', () => {
+    it('should return true for right-to-left strings', () => {
+      expect(isRtl('\u200F\u00E9')).to.eql(true)
+    })
+
+    it('should return false for non right-to-left strings', () => {
+      expect(isRtl('\u00E9')).to.eql(false)
+    })
+  })
+
+  describe('stringDirection', () => {
+    it('should return \'rtl\' for right-to-left strings', () => {
+      expect(stringDirection('\u200F\u00E9')).to.eql('rtl')
+    })
+
+    it('should return null for left-to-right strings', () => {
+      expect(stringDirection('\u00E9')).to.eql(null)
+    })
+  })
+
+  describe('interleave', () => {
+    it('should interleave two arrays', () => {
+      expect(interleave([1,2,3], [4,5,6])).to.eql([1,4,2,5,3,6])
+    })
+
+    it('should drop falsey values', () => {
+      expect(interleave([1,2,null,3,8], [4,5,6,null,7])).to.eql([1,4,2,5,6,3,8,7])
+    })
+
+    it('should include excess values from second argument at the end', () => {
+      expect(interleave([1], [2,3,4])).to.eql([1,2,3,4])
+      expect(interleave([], [1,2,3])).to.eql([1,2,3])
+      expect(interleave([], [])).to.eql([])
+    })
+
+    it('should map notes to correct object structure', () => {
+      expect(
+        interleave(
```

Comment:
This test confused me because I forgot from looking at the name of the function that it is specifically for interleaving parallel data. Should it include "parallel" in the name such as "interleaveParallel"?

Reply:
yeah good point, I've renamed that and expanded the comment.